### ISSUE TITLE
[FS] Cannot decode json message

### ISSUE THREAD

**Issue by CheapHasz** [php-enqueue/enqueue-dev#202]
I have an issue which seems similar to #170, (the error message definitely is) but isn't fixed by #181. 

After consuming a long queue of messages, one fails. Always the same at the first run, and then many after each time I run the consumers again

Error : 
``` 
[LogicException]                                                                                                                                         
  Cannot decode json message '                                                          |{"body":"{\"path\":\"\\\/p\\\/r\\\/pr_swoppad_6_4910_black_1.jpg  
  \",\"filters\":null,\"force\":false}","properties":{"enqueue.topic_name":"liip_imagine_resolve_cache"},"headers":{"content_type":"application\/json","m  
  essage_id":"e94c56dc-8d08-4ba0-8030-6978ca629d97","timestamp":1505379147,"reply_to":null,"correlation_id":""}}'
```

The part of the file before failure :
```
|{"body":"{\"path\":\"\\\/p\\\/r\\\/pr_swoppad_6_4910_red_1.jpg\",\"filters\":null,\"force\":false}","properties":{"enqueue.topic_name":"liip_imagine_resolve_cache"},"headers":{"content_type":"application\/json","message_id":"46fdc345-5d0c-426e-95ac-227c7e657839","timestamp":1505379216,"reply_to":null,"correlation_id":""}}                                                          |{"body":"{\"path\":\"\\\/p\\\/r\\\/pr_swoppad_6_4910_black_1.jpg\",\"filters\":null,\"force\":false}","properties":{"enqueue.topic_name":"liip_imagine_resolve_cache"},"headers":{"content_type":"application\/json","message_id":"c4d60e39-3a8c-42df-b536-c8b7c13e006d","timestamp":1505379216,"reply_to":null,"correlation_id":""}}                                                          |{"body":"{\"path\":\"\\\/p\\\/r\\\/pr_swoppad_6_4910_green_1.jpg\",\"filters\":null,\"force\":false}","properties":{"enqueue.topic_name":"liip_imagine_resolve_cache"},"headers":{"content_type":"application\/json","message_id":"3a6aa176-c879-4435-9626-c48e0643defa","timestamp":1505379216,"reply_to":null,"correlation_id":""}}
```

I really can't see what differs in the failing line

**Comment by makasim** [comment]
The bug was introduced in #181 and unfortunately, it corrupts the queue files so they are no longer useable. #203 fixes the actual bugs and also adds some guards assertions for the message structure so we can spot bad behavior much earlier. 

It is going to be merged to 0.8 branch.

**Comment by CheapHasz** [comment]
Ok great, thanks
